### PR TITLE
Add proposal to change URL section in API Guidelines - SFALG-34

### PR DESCRIPTION
**Pull request by ghijsma**

### CODE REVIEW

File: Guidelines.md [hunk 1]
```diff
@@ -239,6 +241,81 @@ For example, the following is acceptable:
...
+Hostname samples:
+
+```
+services.gentgrp.gent.be
+api.gent.be
+```
+
+Digipolis will always provide the Hostname.
+
+#### 7.1.3 Service namespace
```
Comment (daften) [inline Guidelines.md]: This is very specific for services that will be on the API manager, not necessarily for backend API's. This means this shouldn't go in the general guidelines.

File: Guidelines.md [hunk 2]
```diff
@@ -239,6 +241,81 @@ For example, the following is acceptable:
...
+```
+
+URL Sample:
+```
+/user_management/
+```
+
+Digipolis will always decide the service namespace.
+
+#### 7.1.4 Version
```
Comment (daften) [inline Guidelines.md]: This is already present in chapter 12: versioning.

File: Guidelines.md [hunk 3]
```diff
@@ -239,6 +241,81 @@ For example, the following is acceptable:
...
+```
+/{namespace}/v{version}
+```
+
+URL sample:
+```
+/user_management/v1
+```
+
+#### 7.1.5 Resource reference
```
Comment (daften) [inline Guidelines.md]: This is already present in 9.3

File: Guidelines.md [hunk 4]
```diff
@@ -239,6 +241,81 @@ For example, the following is acceptable:
...
+URL template:
+```
+/{namespace}/v{version}/{resource}/{resource-id}/{sub-resource}/{sub-resource-id}
+```
+
+URL sample:
+```
+/user_management/v1/users/001
+```
+#### 7.1.6 Baseurl
```
Comment (daften) [inline Guidelines.md]: Why would we need to define this, it isn't used anywhere else in the document?

File: Guidelines.md [hunk 5]
```diff
@@ -239,6 +241,81 @@ For example, the following is acceptable:
 https://api.contoso.com/v1/items?url=https://resources.contoso.com/shoes/fancy
 ```
 
+#### 7.1.2 Hostname
```
Comment (daften) [inline Guidelines.md]: This shouldn't be in here, this is simple common sense. This is not an API guideline.

File: Guidelines.md [hunk 6]
```diff
@@ -214,7 +214,9 @@ Clients MAY rely on ordering behavior explicitly identified by the service.
 Clients requesting OPTIONAL server functionality (such as optional headers) MUST be resilient to the server ignoring that particular functionality.
 
 ## 7 Consistency fundamentals
-### 7.1 URL structure
+### 7.1 URL template
+#### 7.1.1 URL construction
```
Comment (daften) [inline Guidelines.md:218]: Why define a 7.1.1 if there is no 7.1.2?

**Review by daften — CHANGES_REQUESTED** [review]
In general the keywords SHOULD, MUST ,... have to be capitalized.

File: Guidelines.md (hunk 4 above)
Comment (ghijsma) [reply Guidelines.md]: I'll remove this as it is not being used elsewhere.

File: Guidelines.md (hunk 1 above)
Comment (ghijsma) [reply Guidelines.md]: To be discussed on the next API workgroup. The whole point of this section is to make sure that the organisation uses the "consistency fundamentals". Service namespace is something we want to introduce for all services. This makes deploying them to the API manager easier. It also makes it easier to put all services into perspective business wise. If necessary we can divide this into two categories of services, the ones deployed on API manager and those that are not?

File: Guidelines.md (hunk 2 above)
Comment (ghijsma) [reply Guidelines.md]: Again here we want to introduce the **MAJOR** version into all API URL's for consistency and better use in the API manager. We may shorten this chapter and make a reference to chapter 12? Also to be discussed on the next API workgroup.

File: Guidelines.md (hunk 5 above)
Comment (ghijsma) [reply Guidelines.md]: I'll also remove this.

File: Guidelines.md (hunk 3 above)
Comment (ghijsma) [reply Guidelines.md]: A collection is the same as a resource? We can squash this together by making the use of this identifier obligatory and setting a reference in 7.1 to 9.3? Again, for consistency and better use in the API manager.

File: Guidelines.md (hunk 6 above)
Comment (ghijsma) [reply Guidelines.md:218]: There was / is? Am going to change 7.1 back to **URL structure** with sub-headers: 

- 7.1.1 URL construction
- 7.1.2 Service namespace
- 7.1.3 Version
- 7.1.4 Resource reference

**Comment by bartlannoeye** [comment]
Besloten op API overleg: Toevoegen als apart chapter, vanuit sectie kan verwezen worden naar de secties.

Voorbeeld business domains (oplijsten in dit chapter):
- supporting
- crm
- location
- finances

**Comment by bartlannoeye** [comment]
Nieuw hoofdstuk kan vlot van master starten, zal deze dus sluiten.

**Pull request closed without merging**